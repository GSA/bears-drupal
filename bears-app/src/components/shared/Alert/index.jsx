import PropTypes from 'prop-types'

const Alert = ({ children }) => {
  return (
    <div className="usa-alert usa-alert--info usa-alert--no-icon">
      <div className="usa-alert__body">
        <p className="usa-alert__text">{children}</p>
      </div>
    </div>
  )
}

Alert.propTypes = {
  children: PropTypes.string,
}

export default Alert
