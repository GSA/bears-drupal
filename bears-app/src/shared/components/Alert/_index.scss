.no-background {
  background-color: #fff !important;
  border-left-color: #fff !important;
}
