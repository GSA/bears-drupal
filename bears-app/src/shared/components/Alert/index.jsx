import PropTypes from 'prop-types'
import { useHandleClassName } from '../../hooks/useHandleClassName'

/**
 * a functional component that renders an information alert
 * @component
 * @param {string} className - inherited classes
 * @param {string} children - inherited children
 * @param {any} alertFieldRef - inherited ref hook
 * @param {string} heading - inherited heading
 * @param {string} description - inherited description
 * @param {bool} error - variant
 * @return {html} returns a wrapped paragraph styled as usa-alert
 */

const Alert = ({
  className,
  children,
  alertFieldRef,
  heading,
  description,
  error,
}) => {
  const defaultClasses = error
    ? ['usa-alert', 'usa-alert--error', 'display-none']
    : ['usa-alert', 'usa-alert--info']

  return (
    <div
      className={useHandleClassName({ className, defaultClasses })}
      role="alert"
      ref={alertFieldRef}
    >
      {children ? (
        <div className="usa-alert__body">
          <p className="usa-alert__text">{children}</p>
        </div>
      ) : (
        <div className="usa-alert__body">
          <h4 className="usa-alert__heading">{heading}</h4>
          <p className="usa-alert__text">{description}</p>
        </div>
      )}
    </div>
  )
}

Alert.propTypes = {
  className: PropTypes.string,
  children: PropTypes.string,
  alertFieldRef: PropTypes.any,
  heading: PropTypes.string,
  description: PropTypes.string,
  error: PropTypes.bool,
}

export default Alert
