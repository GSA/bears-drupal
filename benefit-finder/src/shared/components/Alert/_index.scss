@use '../../styles/colors/_index.scss' as color;

#benefit-finder {
  .no-background {
    background-color: color.$white !important;
    border-left-color: color.$white !important;
  }
}
