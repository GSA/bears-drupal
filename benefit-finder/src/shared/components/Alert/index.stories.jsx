import Alert from './index.jsx'

export default {
  component: Alert,
  tags: ['autodocs'],
  args: {
    children: 'Visit the agency for full eligibility and requirements.',
  },
}

export const Primary = {}
