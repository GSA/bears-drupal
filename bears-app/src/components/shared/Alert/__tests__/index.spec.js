import renderer from 'react-test-renderer'
import Alert from '../index.jsx'

describe('Alert', () => {
  it('renders a match to the previous snapshot', () => {
    const component = renderer.create(<Alert />)
    expect(component.toJSON()).toMatchSnapshot()
  })
})
