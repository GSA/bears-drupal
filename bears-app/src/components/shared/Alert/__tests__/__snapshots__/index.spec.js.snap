// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert renders a match to the previous snapshot 1`] = `
<div
  className="usa-alert usa-alert--info usa-alert--no-icon"
>
  <div
    className="usa-alert__body"
  >
    <p
      className="usa-alert__text"
    />
  </div>
</div>
`;
